docs/guide/advanced/navigation-failures.md

```diff
@@ -0,0 +1,53 @@
+# Navigation Failures
+
+When using `router-link`, Vue Router internally calls `router.push` to trigger a navigation. Depending on the current location and existing [Navigation Guards](./navigation-guards.md), this navigation might end up in a new page being shown, but there are a couple of situations where we will stay on the same page:
+
+- We are already on the page we are trying to go to
+- A navigation guard aborts the navigation by calling `next(false)`
+- An error is thrown in a navigation by calling `next(new Error())`
+
+When using a regular `router-link`, **none of these failures will log an error**. However, if you are using `router.push` or `router.replace`, you might come across an _"Uncaught (in promise) Error"_ message followed by a more specific message in your console. This is part of the _Navigation Failures_ system in Vue Router and it is only available from version 3.2.0 onwards but existed for a long time before: through the two optional callbacks, `onComplete` and `onAbort` that can be passed to `router.push`. Since version 3.1.0, `router.push` and `router.replace` return a _Promise_ if no `onComplete`/`onAbort` callback is provided. This _Promise_ resolves instead of invoking `onComplete` and rejects instead of invoking `onAbort`.
+
+## Detecting Navigation Failures
+
+_Navigation Failures_ are `Error` instances with a few extra properties. Among them, you can find a `type` property. This will allow you to check the type of the navigation failure:
+
+```js
+import { NavigationFailureTypes } from 'vue-router'
```

Comment:
As far as I can see, you're not actually exporting this? Importing it just gives me `undefined` with v3.3.4.

Reply:
This is not yet implemented in vue router 3

Reply:
Okay, I think removing the name before implementing this was a bad idea, but I guess you've gathered as much from the comments in #3047.

I think checking the error type as documented in these docs is fine when `NavigationFailureTypes` is exported. However, as far as I can see, you silently ignore the types you don't check. You should rethrow the error if it's any of the other types.

Reply:
Yeah, I think will add back the `name` for the only error that exposed a `name` before but it won't be documented anywhere.
It's true the examples could show how a non caught error is thrown